packages/astro/src/core/add/index.ts

```diff
@@ -289,11 +289,30 @@ async function parseAstroConfig(configURL: URL): Promise<t.File> {
 	return result;
 }
 
+// Convert an arbitrary NPM package name into a JS identifier
+// Some examples:
```

Comment:
now that we're post v1.0 I'm feeling the burn to start adding some tests to anything like this that we missed for v1.0, but okay to keep kicking this can a bit longer until we're out of the release bug swarm

Reply:
@FredKSchott +1! I vote using Vitest for this for snapshot testing though. The lack of snapshots in `create-astro` made for a _tough_ test authoring experience and brittle tests we ended up removing.